### PR TITLE
[14.0][MIG] account_spread_cost_revenue

### PR DESCRIPTION
**Pull request by kittiu**
Standard migration

### CODE REVIEW

**Comment by kittiu** [comment]
@astirpe I have made the migration of the module out of your PR. Most seem ok. Except here,
https://github.com/OCA/account-financial-tools/pull/1244/files#diff-c57937ecf6269df4b995260fb68911ef84a4c3d11f51ab5aa91f5144d705ea52R40

I need to add this as v13 is allowed to reconciled without posted state, but not in v14.

File: account_spread_cost_revenue/views/account_spread.xml [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+                                    />
+                                    <field name="date" readonly="1" />
+                                    <field name="move_id" readonly="1" />
+                                    <button
+                                        name="create_move"
+                                        icon="fa-play"
+                                        string="Create Move"
+                                        type="object"
+                                        groups="account.group_account_manager"
+                                        attrs="{'invisible':['|',('move_id','!=',False)]}"
```
Comment (astirpe) [inline account_spread_cost_revenue/views/account_spread.xml]:
```suggestion
                                        attrs="{'invisible':[('move_id','!=',False)]}"
```

**Review by astirpe — CHANGES_REQUESTED** [review]
@kittiu I can't test it on Runbot but I tried locally and I got this error:

```
The sequence regex should at least contain the seq grouping keys. For instance:
^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$
```

To reproduce it just click on the "Create Move" button on any spread line

File: account_spread_cost_revenue/views/account_spread.xml (hunk 1 above)
Comment (kittiu) [reply account_spread_cost_revenue/views/account_spread.xml]: Fixed!

**Comment by kittiu** [comment]
May I explain the main point of change on this MIG,

**Odoo 13**
Odoo can reconcile move lines, even on state "draft". And so, the logic and test was also to prove that.
(which actually not quite correct in my opinion). 

**Odoo 14**
Only reconcile after posting document (from both side, invoice and spread move)

* https://github.com/OCA/account-financial-tools/pull/1244/commits/470d872ff65a797aa6aa308a79d2334c577a7bb0#diff-c57937ecf6269df4b995260fb68911ef84a4c3d11f51ab5aa91f5144d705ea52L38-L39
* https://github.com/OCA/account-financial-tools/pull/1244/commits/470d872ff65a797aa6aa308a79d2334c577a7bb0#diff-50788f2526bc0ec75ae627daa97d005c29444834c499e006b0ea77f79d57d6a9R501-R502
* https://github.com/OCA/account-financial-tools/pull/1244/commits/470d872ff65a797aa6aa308a79d2334c577a7bb0#diff-e279bec9360828ba2ebeb82a3b26b82424e049b4a6e547c1d74aa5cbaa7b81c4R16-R20

**Comment by kittiu** [comment]
Looks like new account numbering of v14 cause problem. When create_move() this module use Description from spread table as Number of Journal Entry. And it always conflict with the pattern of v14.

May be we need to use odoo numbering system (MISC/...) instead?

@astirpe WDYT?

**Comment by astirpe** [comment]
@kittiu yes, I agree

**Comment by kittiu** [comment]
> @kittiu yes, I agree

Fixed.

**Comment by astirpe** [comment]
I made a quick test locally and it seems working fine! I would like to spend some extra time to test more in deep, unless there are other contributors who want to do that.

**Review by ps-tubtim — APPROVED** [review]
LGTM.

**Review by Saran440 — APPROVED** [review]
LGTM :+1:

**Comment by hitrosol** [comment]
@kittiu, I can't test this PR in on runboat since it still refer to runbot instead of runboat. 
It functionally works well in my local. Can we merge this ?

**Review by espo-tony — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

**Comment by Saran440** [comment]
@kittiu you use wrong typo command. please do it again.

**Review by luc-demeyer — APPROVED** [review]

**Comment by luc-demeyer** [comment]
Code review & testing ok.
Pls merge.

File: account_spread_cost_revenue/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,10 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
```
Comment (luc-demeyer) [inline account_spread_cost_revenue/security/ir.model.access.csv:1]: read access should be added for account.group_account_invoice

**Review by luc-demeyer — CHANGES_REQUESTED** [review]
read access must added for the spread models for account.group_account_invoice

**Comment by StefanRijnhart** [comment]
@kittiu would you like to address the permission comment by @luc-demeyer?

**Comment by Saran440** [comment]
@StefanRijnhart @luc-demeyer Fixed it. can you check please?

**Comment by StefanRijnhart** [comment]
Looking good! Can you squash the new commit into the migration commit?

**Review by StefanRijnhart — APPROVED** [review]
Thanks for the quick follow-up! @luc-demeyer is it OK for you now?

**Comment by Saran440** [comment]
@StefanRijnhart @luc-demeyer  Hi, let's merge this pull request. We will continue to migrate this module to version 15.

**Comment by StefanRijnhart** [comment]
Agreed!

/ocabot merge nobump

**Pull request merged**